token/options.go

```diff
@@ -84,6 +87,43 @@ func WithSSH(v interface{}) Options {
 	})
 }
 
+// WithConfirmationKid returns an Options function that sets the cnf claim with
+// the given kid.
+func WithConfirmationKid(kid string) Options {
+	return func(c *Claims) error {
+		c.Set(ConfirmationClaim, map[string]string{
+			"kid": kid,
+		})
+		return nil
+	}
+}
+
+// WithFingerprint returns an Options function that the cnf claims with the kid
+// representing the fingerprint of the certificate request or the ssh public
+// key.
+func WithFingerprint(v interface{}) Options {
+	return func(c *Claims) error {
+		var data []byte
+		switch vv := v.(type) {
+		case *x509.CertificateRequest:
+			data = vv.Raw
+		case ssh.PublicKey:
+			data = vv.Marshal()
+		default:
+			return fmt.Errorf("unsupported fingerprint for %T", vv)
+		}
+
+		kid, err := fingerprint.New(data, crypto.SHA256, fingerprint.Base64RawURLFingerprint)
+		if err != nil {
+			return err
+		}
+		c.Set(ConfirmationClaim, map[string]string{
+			"kid": kid,
+		})
```

Comment:
For the X509 case, the value maybe shouldn't be put in `kid`, as we generally use that claim to carry the thumbprint of a JWK (used as key identifier); not the hash of a full request. Arguably, the `kid` can be filled arbitrarily, as long as it identifiers the key, so it's not wrong to put the hash of the CSR in there, but I think using a different name might be better.

There's `"cnf":{"x5t#S256":"...."}` in https://datatracker.ietf.org/doc/html/rfc8705#section-appendix.a, which is for certificates. Can we find/use/create a (custom) variant of that for certificate requests? E.g. `"x5rt#S256"`, or something like that?

Reply:
Fixed with eb2564